### PR TITLE
Query request generation

### PR DESCRIPTION
**Pull request by jakobbraun**
Automatic generation for DynamoDB query requests.

### CODE REVIEW

File: src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/AbstractDynamodbIndex.java [hunk 1]
```diff
@@ -4,20 +4,20 @@
 import java.util.Optional;
 
 /**
- * This class represents a DynamoDB index. This can either be the default index of a table or a secondary index.
+ * This class represents a DynamoDB index. This can either be the primary index of a table or a secondary index.
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/AbstractDynamodbIndex.java:7]: I think the description should be a little bit different for an abstract class. Something like "This class is an abstract base for..."

File: src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadata.java [hunk 2]
```diff
@@ -71,8 +72,9 @@ public DynamodbIndex getPrimaryIndex() {
      * 
      * @return list of all keys
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadata.java:73]:
```suggestion
     * @return list of all indexes
```

File: src/main/java/com/exasol/adapter/dynamodb/queryrunner/DynamodbQueryIndexSelector.java [hunk 3]
```diff
@@ -24,19 +24,19 @@
      * @param availableIndexes list of possible DynamoDB indexes
      * @return most restricted index or null if non did match the criteria
      */
-    public DynamodbIndex findMostRestrictedIndex(final QueryPredicate<DynamodbNodeVisitor> selection,
-            final List<DynamodbIndex> availableIndexes) {
-        final List<DynamodbIndex> suitableKeys = extractSuitableIndexes(selection, availableIndexes);
+    public AbstractDynamodbIndex findMostRestrictedIndex(final QueryPredicate<DynamodbNodeVisitor> selection,
+            final List<AbstractDynamodbIndex> availableIndexes) {
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/dynamodb/queryrunner/DynamodbQueryIndexSelector.java:28]: I don't really like that you have a list of abstract class's descendants. Maybe you can also create an interface for these classes?

File: src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadata.java [hunk 4]
```diff
@@ -9,9 +9,9 @@
  */
 public class DynamodbTableMetadata implements Serializable {
     private static final long serialVersionUID = 7898591572006230212L;
-    private final DynamodbIndex primaryIndex;
-    private final List<DynamodbIndex> localIndexes;
-    private final List<DynamodbIndex> globalIndexes;
+    private final DynamodbPrimaryIndex primaryIndex;
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadata.java:12]: You can use interface here, I think. I like it somehow more as it was before changes

File: src/main/java/com/exasol/adapter/dynamodb/queryrunner/DynamodbQueryPlanner.java [hunk 5]
```diff
@@ -21,7 +21,7 @@ public DynamodbQueryPlan planQuery(final RemoteTableQuery<DynamodbNodeVisitor> d
             final DynamodbTableMetadata tableMetadata) {
         final String tableName = documentQuery.getFromTable().getRemoteName();
         try {
-            return new DynamodbGetItemQueryPlanFactory().buildGetItemPlanIfPossible(documentQuery, tableMetadata);
+            return new DynamodbQueryQueryPlanFactory().buildQueryPlanIfPossible(documentQuery, tableMetadata);
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/dynamodb/queryrunner/DynamodbQueryPlanner.java:24]: Maybe rename to `DynamodbQueryPlanFactory`?

File: src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadata.java (hunk 4 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadata.java:12]: But the primary index is always a `DynamodbPromaryIndex`. Using  the interface here would loose the type information.

**Pull request merged**